src/main/java/com/redhat/devtools/intellij/tektoncd/ui/editors/PipelineGraphUpdater.java

```diff
@@ -0,0 +1,167 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.intellij.tektoncd.ui.editors;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.mxgraph.view.mxGraph;
+import io.fabric8.tekton.pipeline.v1beta1.Pipeline;
+import io.fabric8.tekton.pipeline.v1beta1.PipelineTask;
+import io.fabric8.tekton.pipeline.v1beta1.PipelineTaskCondition;
+import io.fabric8.tekton.pipeline.v1beta1.PipelineTaskInputResource;
+import org.apache.commons.lang.StringUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+
+public class PipelineGraphUpdater implements GraphUpdater<Pipeline> {
+    enum Type {
+        TASK,
+        CONDITION;
+    }
+
+    static class Node {
+        Type type;
+        String id;
+        String name;
+        Collection<Node> childs = new ArrayList();
+
+        public Node(Type type, String id, String name) {
+            this.type = type;
+            this.id = id;
+            this.name = name;
+        }
+    }
+    public static final String TASK_PREFIX = "Task:";
+    public static final String CONDITION_PREFIX = "Condition:";
+    public static final String FINALLY_PREFIX= "Finally:";
+    private final ObjectMapper MAPPER  = new ObjectMapper(new YAMLFactory());
+
+    @Override
+    public Pipeline adapt(String content) {
+        try {
+            return MAPPER.readValue(content, Pipeline.class);
+        } catch (IOException e) {
+            return null;
+        }
+    }
+
+    @Override
+    public void update(Pipeline content, mxGraph graph) {
+        List<PipelineTask> tasks = content.getSpec().getTasks();
```

Comment:
Here we should check if spec section exists otherwise we can have an error. 
If you create a new pipeline, delete the spec section and start by changing the name you have 
![image](https://user-images.githubusercontent.com/49404737/92378422-10ab9700-f106-11ea-9830-f86f70bc8289.png)

Reply:
Can't reproduce the move block issue, I tested both on Windows and Linux but noticed that when you move a condition, the connector is not correctly placed. See 
![tekton1](https://user-images.githubusercontent.com/695993/92399168-1e731380-f12a-11ea-876c-e08d2e950ce8.gif)
Will provide a fix for the empty spec case.

Reply:
I tried it again and incredibly i am not able to replicate it, even by changing theme, now it works .... don't know why but that's good lol